academy/exchange/__init__.py

```diff
@@ -74,11 +98,12 @@ def _create_transport(
         mailbox_id: EntityId | None = None,
         *,
         name: str | None = None,
+        registration_info: RegistrationInfo[BehaviorT] | None = None,
     ) -> ExchangeTransport: ...
 
     def create_agent_client(
         self,
-        agent_id: AgentId[BehaviorT],
+        agent_info: RegistrationInfo[BehaviorT],
```

Comment:
I assume that a `FooExchangeFactory.create_agent_client()` will need to take the exact `RegistrationInfo` type produced by `FooExchangeTransport.register_agent()`. We'll probably need to make `ExchangeTransport` and `ExchangeFactory` generic on the `RegistrationInfo` type then otherwise mypy is going to complain as soon as we add something that's not `BasicRegistrationInfo`.